tests/serde.rs

```diff
@@ -3,9 +3,144 @@ use std::{collections::HashMap, fs::File, path::Path};
 
 use gvas::GvasFile;
 
-fn get_hints() -> HashMap<String, String> {
+fn test_file_with_hints(path: &str, hints: &HashMap<String, String>) {
+    let path = Path::new(env!("CARGO_MANIFEST_DIR")).join(path);
+    let mut file = File::open(path).expect("Open test asset");
+    let file =
+        GvasFile::read_with_hints(&mut file, GameVersion::Default, hints).expect("Parse gvas file");
+    let value = serde_json::to_string(&file).expect("Deserialize");
+    let from_value = serde_json::from_str::<GvasFile>(value.as_str()).expect("Serialize");
+    assert_eq!(file, from_value);
+}
+
+fn test_file(path: &str) {
+    test_file_with_hints(path, &HashMap::new());
+}
+
+#[test]
+fn serde_assert_failed() {
+    test_file("resources/test/assert_failed.sav");
+}
+
+// #[test]
```

Comment:
maybe we should just `#[ignore]` it in this case, instead of commenting the code out?